src/meta/src/barrier/notifier.rs

```diff
@@ -22,11 +22,14 @@ pub(super) struct Notifier {
     /// Get notified when scheduled barrier is about to send.
     pub to_send: Option<oneshot::Sender<()>>,
 
-    /// Get notified when scheduled barrier is collected or failed.
-    pub collected: Option<oneshot::Sender<MetaResult<()>>>,
+    /// Get notified when scheduled barrier(checkpoint = true) is collected or failed,
+    pub collected_checkpoint: Option<oneshot::Sender<MetaResult<()>>>,
 
     /// Get notified when scheduled barrier is finished.
     pub finished: Option<oneshot::Sender<()>>,
+
+    /// Get notified when scheduled barrier(checkpoint = false) is collected or failed.
+    pub collected_no_checkpoint: Option<oneshot::Sender<MetaResult<()>>>,
```

Comment:
Will this be triggered if barrier(checkpoint = true) is collected? If yes, I suggest rename the two newly added fields to collected_checkpoint_barrier and collected_barrier.

Reply:
and is it better to use a enum to wrap these two types of sender?